drivers/pinmux/dev/pinmux_dev_stm32.c

```diff
@@ -1,81 +0,0 @@
-/*
- * Copyright (c) 2016 Open-RnD Sp. z o.o.
- *
- * SPDX-License-Identifier: Apache-2.0
- */
-
-/**
- * @brief
- *
- * A common driver for STM32 pinmux. Each SoC must implement a SoC
- * specific part of the driver.
- */
-
-#include <errno.h>
-
-#include <kernel.h>
-#include <device.h>
-#include <soc.h>
-#include "pinmux.h"
-#include <pinmux.h>
-#include <pinmux/stm32/pinmux_stm32.h>
-
-static int pinmux_stm32_set(struct device *dev,
-				 u32_t pin, u32_t func)
-{
-	ARG_UNUSED(dev);
-
-	return _pinmux_stm32_set(pin, func, NULL);
-}
-
-static int pinmux_stm32_get(struct device *dev,
-				 u32_t pin, u32_t *func)
-{
-	ARG_UNUSED(dev);
-	ARG_UNUSED(pin);
-	ARG_UNUSED(func);
-
-	return -ENOTSUP;
-}
-
-static int pinmux_stm32_input(struct device *dev,
-				u32_t pin,
-				u8_t func)
-{
-	ARG_UNUSED(dev);
-	ARG_UNUSED(pin);
-	ARG_UNUSED(func);
-
-	return -ENOTSUP;
-}
-
-static int pinmux_stm32_pullup(struct device *dev,
-				u32_t pin,
-				u8_t func)
-{
-	ARG_UNUSED(dev);
-	ARG_UNUSED(pin);
-	ARG_UNUSED(func);
-
-	return -ENOTSUP;
-}
-
-static const struct pinmux_driver_api pinmux_stm32_api = {
```

Comment:
What is pinmux-dev exactly? Is it this API? If so it's also provided by most SoCs, just in the parent directory.

Reply:
was a split of pinmux APIs to enable some development related APIs that would not be used directly by users, i.e. to avoid someone from bricking their board etc. It is still there for historical reasons, should have been removed long ago.